### ISSUE TITLE
Custom attribute in fhir enpoint to fhir resource should not be in first-class attribute

### ISSUE THREAD

**Issue by jyeros** [Aidbox/Issues#147]
I create a custom attribute to a fhir resource and then post the one by aidbox endpoint. 

Get resource from aidbox endpoint all works good, the attribute is showed as first-class one.

Get resource from fhir endpoint the attribute is showed as first-class attribute too and it should be in extension array instead. For example
```
Post /Attribute

{
  "path": [
    "temp"
  ],
  "module": "phyl2",
  "description": "temp",
  "id": "CatalogEntry.temp",
  "resource": {
    "id": "CatalogEntry",
    "resourceType": "Entity"
  },
  "type": {
    "id": "Reference",
    "resourceType": "Entity"
  },
  "refers": [
    "Organization"
  ],
  "resourceType": "Attribute",
  "isCollection": true
}
```
```
Post /CatalogEntry

{
    "type": {
        "text": "service"
    },
    "status": "active",
    "orderable": false,
    "identifier": [
        {
            "use": "official",
            "value": "34",
            "system": "http://test.test"
        }
    ],
    "referencedItem": {
        "id": "bbc4c7f8-a9c9-423c-a359-81c7d412ba91",
        "resourceType":"ObservationDefinition"
    },
    "resourceType": "CatalogEntry",
    "temp":[
        {
            "id": "620cef1c-1e63-4b69-9d49-52a024eaa141",
            "resourceType": "Organization"
        }
    ]
}
```
aidbox endpoint ✅ 
```
Get /CatalogEntry/9bfcce2a-3fc6-4c52-9235-b697a4cb98fc

temp:
- {id: 620cef1c-1e63-4b69-9d49-52a024eaa141, resourceType: Organization}
meta: {lastUpdated: '2019-08-30T04:27:49.334Z', versionId: '1931'}
referencedItem: {id: bbc4c7f8-a9c9-423c-a359-81c7d412ba91, resourceType: ObservationDefinition}
type: {text: service}
resourceType: CatalogEntry
orderable: false
status: active
id: 9bfcce2a-3fc6-4c52-9235-b697a4cb98fc
identifier:
- {use: official, value: '34', system: 'http://test.test'}
```
fhir endpoint ❌ 
```
Get /fhir/CatalogEntry/9bfcce2a-3fc6-4c52-9235-b697a4cb98fc

temp:
- {id: 620cef1c-1e63-4b69-9d49-52a024eaa141, resourceType: Organization}
meta: {lastUpdated: '2019-08-30T04:27:49.334Z', versionId: '1931'}
referencedItem: {reference: ObservationDefinition/bbc4c7f8-a9c9-423c-a359-81c7d412ba91}
type: {text: service}
resourceType: CatalogEntry
orderable: false
status: active
id: 9bfcce2a-3fc6-4c52-9235-b697a4cb98fc
identifier:
- {use: official, value: '34', system: 'http://test.test'}
```
fhir endpoint should be 
```
Get /fhir/CatalogEntry/9bfcce2a-3fc6-4c52-9235-b697a4cb98fc

meta: {lastUpdated: '2019-08-30T04:27:49.334Z', versionId: '1931'}
referencedItem: {reference: ObservationDefinition/bbc4c7f8-a9c9-423c-a359-81c7d412ba91}
type: {text: service}
resourceType: CatalogEntry
orderable: false
status: active
id: 9bfcce2a-3fc6-4c52-9235-b697a4cb98fc
identifier:
- {use: official, value: '34', system: 'http://test.test'}
extension:
- url: temp
  valueReference: {reference: Organization/620cef1c-1e63-4b69-9d49-52a024eaa141}
```

**Comment by ruscoder** [comment]
@jyeros Hello! You need to specify `extensionUrl` property in the Attribute definition.

Try:
```
Post /Attribute

{
  "path": [
    "temp"
  ],
  "module": "phyl2",
  "description": "temp",
  "id": "CatalogEntry.temp",
  "resource": {
    "id": "CatalogEntry",
    "resourceType": "Entity"
  },
  "type": {
    "id": "Reference",
    "resourceType": "Entity"
  },
  "refers": [
    "Organization"
  ],
  "resourceType": "Attribute",
  "isCollection": true,
  "extensionUrl": " https://test.test/extension/temp"
}
```

**Comment by jyeros** [comment]
Thanks @ruscoder I specify the property and works. But i am having the problem related with #135 now in fhir extension array because the `valueReference` is `id` and `resourceType` not `reference`.

**Comment by niquola** [comment]
Should be fixed in 0.4.7-SNAPSHOT (build is coming in 5 min)
